llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -542,6 +548,15 @@ std::string saveModuleProperties(module_split::ModuleDesc &MD,
     PropSet.add(PropSetRegTy::SYCL_HOST_PIPES, HostPipePropertyMap);
   }
 
+  if (MD.isSpecConstantDefault()) {
+    PropSet[PropSetRegTy::SYCL_MISC_PROP].insert({"defaultSpecConstants", 1});
+    const std::optional<std::string> &OriginalImageName =
+        MD.tryGetOriginalImageName();
+    assert(OriginalImageName.has_value() && "Original image name must be set");
+    PropSet[PropSetRegTy::SYCL_MISC_PROP].insert(
+        {"originalImage", StringRef(*OriginalImageName)});
```

Comment:
Why do we need that? Can't runtime just select any other device image, which is compatible with the device and has necessary kernels?

Please note that device image names do not leave `sycl-post-link`, they are not available to runtime and therefore this property is useless there.

Having the first property you add should be enough for RT. We could even rename it to something like `nonModifiableSpecConstants`